### PR TITLE
PP-509 remove tokens from acceptedTokens array

### PR DESCRIPTION
**Pull request by ironFe93**
## What

- implement removeToken()

## Why

- A method to delist tokens from rif-relay was needed

### CODE REVIEW

**Comment by franciscotobar** [comment]
Im not sure if this feature needs to be included on this branch since its not related with ethers.

**Comment by franciscotobar** [comment]
I did some testing. If I send the address with a different index, the token removed is the one from the index and the validation seems to be that is not working.

**Comment by ironFe93** [comment]
mmm I will attempt to do the same, thanks for reviewing btw!

**Comment by ironFe93** [comment]
@franciscotobar I updated the implementation, what you showed me indicated that one more validation was required, so I included it and a corresponding test.

**Review by franciscotobar — APPROVED** [review]
LGTM

File: contracts/verifier/CustomSmartWalletDeployVerifier.sol [hunk 1]
```diff
@@ -89,6 +89,15 @@ contract CustomSmartWalletDeployVerifier is
         acceptedTokens.push(token);
     }
 
+    function removeToken(address token, uint256 index) external onlyOwner {
+        require(token != address(0), "Token cannot be zero address");
+        require(tokens[token] == true, "Token is not accepted");
```
Comment (franciscotobar) [inline contracts/verifier/CustomSmartWalletDeployVerifier.sol:94]:
```suggestion
        require(tokens[token], "Token is not accepted");
```

File: contracts/verifier/DeployVerifier.sol [hunk 2]
```diff
@@ -92,6 +92,15 @@ contract DeployVerifier is IDeployVerifier, ITokenHandler, Ownable {
         return acceptedTokens;
     }
 
+    function removeToken(address token, uint256 index) external onlyOwner {
+        require(token != address(0), "Token cannot be zero address");
+        require(tokens[token] == true, "Token is not accepted");
```
Comment (franciscotobar) [inline contracts/verifier/DeployVerifier.sol:97]:
```suggestion
        require(tokens[token], "Token is not accepted");
```

File: contracts/verifier/RelayVerifier.sol [hunk 3]
```diff
@@ -85,6 +85,15 @@ contract RelayVerifier is IRelayVerifier, ITokenHandler, Ownable {
         acceptedTokens.push(token);
     }
 
+    function removeToken(address token, uint256 index) external onlyOwner {
+        require(token != address(0), "Token cannot be zero address");
+        require(tokens[token] == true, "Token is not accepted");
```
Comment (franciscotobar) [inline contracts/verifier/RelayVerifier.sol:90]:
```suggestion
        require(tokens[token], "Token is not accepted");
```

**Pull request merged**